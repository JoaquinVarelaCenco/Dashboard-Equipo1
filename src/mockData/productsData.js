module.exports = [
    {
      "id": 2,
      "title": "iPhone X",
      "description": "SIM-Free, Model A19211 6.5-inch Super Retina HD display with OLED technology A12 Bionic chip with ...",
      "price": 899,
      "rating": { "rate": 4.44, "count": 126 },
      "stock": 34,
      "category": "smartphones",
      "images": [],
      "lastModified": "2022-10-12T01:07:11.630Z"
    },
    {
      "id": 3,
      "title": "Samsung Universe 9",
      "description": "Samsung's new variant which goes beyond Galaxy to the Universe",
      "price": 1249,
      "rating": { "rate": 4.09, "count": 845 },
      "stock": 36,
      "category": "smartphones"
    },
    {
      "id": 23,
      "title": "Orange Essence Food Flavou",
      "description": "Specifications of Orange Essence Food Flavour For Cakes and Baking Food Item",
      "price": 14,
      "rating": { "rate": 4.85, "count": 1645 },
      "stock": 26,
      "category": "groceries",
      "images": [
        "https://dummyjson.com/image/i/products/23/1.jpg",
        "https://dummyjson.com/image/i/products/23/2.jpg",
        "https://dummyjson.com/image/i/products/23/3.jpg",
        "https://dummyjson.com/image/i/products/23/4.jpg",
        "https://dummyjson.com/image/i/products/23/thumbnail.jpg"
      ]
    },
    {
      "id": 24,
      "title": "cereals muesli fruit nuts",
      "description": "original fauji cereal muesli 250gm box pack original fauji cereals muesli fruit nuts flakes breakfast cereal break fast faujicereals cerels cerel foji fouji",
      "price": 46,
      "rating": { "rate": 4.94, "count": 11378 },
      "stock": 113,
      "category": "groceries",
      "images": [
        "https://dummyjson.com/image/i/products/24/1.jpg",
        "https://dummyjson.com/image/i/products/24/2.jpg",
        "https://dummyjson.com/image/i/products/24/3.jpg",
        "https://dummyjson.com/image/i/products/24/4.jpg",
        "https://dummyjson.com/image/i/products/24/thumbnail.jpg"
      ],
      "lastModified": "2022-10-17T14:31:36.884Z"
    },
    {
      "id": 25,
      "title": "Gulab Powder 50 Gram",
      "description": "Dry Rose Flower Powder Gulab Powder 50 Gram â€¢ Treats Wounds",
      "price": 70,
      "rating": { "rate": 4.87, "count": 568 },
      "stock": 47,
      "category": "groceries",
      "images": [
        "https://dummyjson.com/image/i/products/25/1.png",
        "https://dummyjson.com/image/i/products/25/2.jpg",
        "https://dummyjson.com/image/i/products/25/3.png",
        "https://dummyjson.com/image/i/products/25/4.jpg",
        "https://dummyjson.com/image/i/products/25/thumbnail.jpg"
      ]
    }
  ]
  