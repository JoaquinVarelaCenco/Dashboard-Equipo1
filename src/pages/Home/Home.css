/* Home css */

.home-charts{
    display: flex;
    width: 100%;
    min-height: 100vh;
    padding: 20px;
}

.home-charts__bar{
    width: 50%;
    height: 500px;
    padding: 10px;
}

.home-charts__pie{
    width: 50%;
    height: 500px;
}

@media (max-width: 830px){
    .home-charts{
        flex-direction: column;
        justify-content: center;
    }
    .home-charts__bar{
        width: 100%;
        height: 400px;
    }
    
    .home-charts__pie{
        width: 100%;
        height: 400px;
    }
}