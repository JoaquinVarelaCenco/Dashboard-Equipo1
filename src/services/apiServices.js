const url = "http://localhost:5000/api/";
const getProducts = () => {
  return fetch(`${url}product`).then((res) => res.json())
};

const getProductById = (id) => {
  return fetch(`${url}product/${id}`).then((res) => res.json());
};

const deleteProduct = (productId) => {
  return fetch(`${url}product?id=${productId}`, {
    method: "DELETE",
  }).then((res) => res.json());
};

const updateProduct = (body) => {
  return fetch(`${url}product`, {
    method: "PUT",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(body),
  }).then((res) => res.json());
};

const newProduct = (body) => {
  return fetch(`${url}product`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(body),
  }).then((res) => res.json());
};

module.exports = {
  getProducts,
  getProductById,
  deleteProduct,
  updateProduct,
  newProduct,
};
