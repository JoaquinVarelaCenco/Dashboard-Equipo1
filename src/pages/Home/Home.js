import React, { useState, useEffect } from 'react';
import { getProducts } from '../../services/apiServices';
import HomeCard from '../../components/HomeCard/HomeCard';

import './Home.css'
import box from '../../assets/images/package-variant-closed.svg';
import store from '../../assets/images/store.svg';
import ChartBar from '../../components/ChartBar/ChartBar';
import PieCharts from '../../components/ChartPie/ChartPie';

const Home = () => {

  const [ totalProducts, setTotalProducts ] = useState(0);
  const [products, setProducts] = useState([]);

  useEffect(() => {
      getProducts()
      .then(res =>{
        setProducts(res)
        setTotalProducts(res.length);
      })
      .catch(error =>{
        console.log(error);
      })
  }, [])
  

  products.sort((a,b) =>{
    if(a.rating.rate < b.rating.rate){
      return 1;
    }else  if(a.rating.rate > b.rating.rate){
      return -1;
    }else{
      return 0;
    }
  });

  const eightMostRate = products.slice(0,8);

  products.sort((a,b) =>{
    if(a.stock < b.stock){
      return 1;
    }else  if(a.stock > b.stock){
      return -1;
    }else{
      return 0;
    }
  });

  const eightMostStock = products.slice(0,8);

  return (
      <div className='home-container'>

        <div className='home-cards'>
          <HomeCard 
            count={totalProducts}
            title={"Productos"}
            linkBtnList= {"/products"}
            textBtnAdd = {"Agregar Producto"}
            linkBtnAdd = {"/products/new"}
            img = {box}
          />

          <HomeCard 
            count={12}
            title={"Tiendas"}
            linkBtnList= {"/stores"}
            textBtnAdd = {"Agregar Tiendas"}
            linkBtnAdd = {"/stores/new"}
            img = {store}
          />
        </div>

        <div className='home-charts'>

          <div className='home-charts__bar'>
            <h2>Productos con mayor Stock</h2>
              <ChartBar 
                products = {eightMostStock}
              />
          </div>

          <div className='home-charts__pie'>
            <h2>Productos con mayor Rating</h2>
              <PieCharts 
                products = {eightMostRate}
              />
          </div>

        </div>

      </div>
  )
}

export default Home