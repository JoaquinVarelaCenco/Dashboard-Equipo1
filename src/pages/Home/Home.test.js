import { act, render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Home from './Home.js'
import { getProducts } from "../../services/apiServices";
import mockData from '../../mockData/productsData';



//mocks
jest.mock('../../services/apiServices');

let component;
let products;


describe("Tests para la vista Home", () => {

  beforeEach( async()=>{
    products = await getProducts.mockResolvedValue([...mockData ])

    await act( async ()=>{
      component = render(
        
        <Home />
        
        , { wrapper: BrowserRouter })
    })
  })


  it("La vista debe renderizarse correctamente", async () => {
    const { container } = component;

    expect(container).toMatchSnapshot()
  });


  it("Se deben renderizar dos componentes HomeCard", ()=>{

    const imagesInView = screen.getAllByRole('img', name="Icono de Card");

    expect(imagesInView.length).toBe(2);
    
  })


});
