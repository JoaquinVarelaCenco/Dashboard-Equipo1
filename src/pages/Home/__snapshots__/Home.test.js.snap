// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests para la vista Home La vista debe renderizarse correctamente 1`] = `
<div>
  <div
    class="home-container"
  >
    <div
      class="home-cards"
    >
      <div
        class="itemRow"
      >
        <div
          class="itemRow__count"
        >
          <img
            alt="Icono de Card"
            src="package-variant-closed.svg"
          />
          <p>
            <strong>
              5
            </strong>
             
            Productos
          </p>
        </div>
        <div
          class="itemRow__buttons"
        >
          <a
            href="/products"
          >
            <button>
              Ver listado
            </button>
          </a>
          <a
            href="/products/new"
          >
            <button>
              Agregar Producto
            </button>
          </a>
        </div>
      </div>
      <div
        class="itemRow"
      >
        <div
          class="itemRow__count"
        >
          <img
            alt="Icono de Card"
            src="store.svg"
          />
          <p>
            <strong>
              12
            </strong>
             
            Tiendas
          </p>
        </div>
        <div
          class="itemRow__buttons"
        >
          <a
            href="/stores"
          >
            <button>
              Ver listado
            </button>
          </a>
          <a
            href="/stores/new"
          >
            <button>
              Agregar Tiendas
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      class="home-charts"
    >
      <div
        class="home-charts__bar"
      >
        <h2>
          Productos con mayor Stock
        </h2>
      </div>
      <div
        class="home-charts__pie"
      >
        <h2>
          Productos con mayor Visitas
        </h2>
      </div>
    </div>
  </div>
</div>
`;
